components/eam/src/physics/cam/microp_aero.F90

```diff
@@ -646,10 +658,35 @@ subroutine microp_aero_run ( &
     
    call outfld('LCLOUD', lcldn, pcols, lchnk)
 
+   ! Init qqcw (a flat 1d pointer array) to mode number and species mass mixing ratios for
+   ! cloud borne phases
+
+   allocate(qqcw(ncnst_tot)) 
+   do imode = 1, ntot_amode
+      do lspec = 0, nspec_amode(imode)  ! loop through all species for mode 'imode'
+          mm = mam_idx(imode,lspec)
+          if (lspec == 0) then   ! number
+             icnst = numptrcw_amode(imode)
+          else ! aerosol mass
+             icnst = lmassptrcw_amode(lspec,imode)
+          endif
+          qqcw(mm)%fld => qqcw_get_field(pbuf,icnst,lchnk,.true.) 
+      enddo
+   enddo
+
+   ! NOTE FOR C++ porting, the following variable obtained using pbuf is used
+   ! for vertical mixing in the activation subroutine dropmixnuc
+   ! Do not use 'kvh_idx' since that index is not necessarily set in this module
+
+   kvh_idx_dropmixnuc      = pbuf_get_index('kvh')
```

Comment:
This can be moved to `microp_aero_init` with `kvh_idx_dropmixnuc` declared at the module level (like `cldo_idx`)

Reply:
OK, done.